aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,729 @@
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Rendering HTML
+--------------
+
+The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+
+.. code-block:: none
+
+  <p>Hello World</p>
+
+Is rendered unchanged as ``<p>Hello World</p>`` by the server.
+
+Rendering server code
+----------------------
+
+Razor supports C# and uses the ``@`` symbol to transition from HTML to C#.
+Razor can transition from HTML into C# or into Razor specific markup. When an ``@`` symbol is followed by a :ref:`Razor reserved keyword <Razor-reserved-keywords-label>` it transitions into Razor specific markup, otherwise it transitions into plain C# .
+
+HTML containing ``@`` symbols may need to be escaped with a second ``@`` symbol. For example:
+
+.. code-block:: none
+
+ <p>@@Username</p>
+
+would render the following HTML:
+
+.. code-block:: none
+
+ <p>@Username</p>
+
+.. _razor-email-ref:
+
+``@`` in an email alias
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+HTML attributes containing email addresses don’t treat the ``@`` symbol as a transition character.
+
+ ``<a href="mailto:Support@contoso.com">Support@contoso.com</a>``
+
+Implicit Razor expressions
+---------------------------
+
+Implicit Razor expressions start with ``@`` followed by C# code. For example:
+
+.. code-block:: none
+
+  <p>@DateTime.Now</p>
+  <p>@DateTime.IsLeapYear(2016)</p>
+
+Implicit expressions generally cannot contain spaces. For example, in the code below, one week is not subtracted from the current time:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @* End of greeting *@
+  :end-before: @*Add () to get correct time.*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>
+    Last week: 7/7/2016 4:39:52 PM - TimeSpan.FromDays(7)
+  </p>
+
+Adding parenthesis fixes the problem:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @*Add () to get correct time.*@
+  :end-before: @*End of correct time*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>
+    Last week: 6/30/2016 4:39:52 PM
+ </p>
+
+With the exception of the C# ``await`` keyword implicit expressions must not contain spaces. For example, you can intermingle spaces as long as the C# statement has a clear ending:
+
+.. code-block:: none
+
+  <p>@await DoSomething("hello", "world")</p>
+
+Explicit Razor expressions
+----------------------------
+
+Explicit Razor expressions consists of an @ symbol with balanced parenthesis. For example, to render last weeks’ time:
+
+.. code-block:: none
+
+  <p>Last week this time: @(DateTime.Now - TimeSpan.FromDays(7))</p>
+
+Any content within the @() parenthesis is evaluated and rendered to the output.
+
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
```

Comment:
We haven't really explained with `IHtmlContent` is about at this point. I suggest starting with the following text:

Any C# expression that evaluates to a string is HTML encoded and rendered to the output HTML. If the expression evaluates to something that implements `IHtmlContent` it is rendered directly to the output by calling `IHtmlContent.WriteTo`. If the expression evaluates to anything else it is converted to a string by calling `ToString`, encoded and rendered to the output.